tofu/modules/generic/test_environment/variables.tf

```diff
@@ -23,6 +23,14 @@ variable "network_config" {
   type        = any
 }
 
+# Only used for Harvester module atm
+variable "image_id" {
+  description = "ID of a Harvester image, if one was created. Otherwise null"
+  type        = string
+  default     = null
+  nullable    = true
+}
```

Comment:
This only being used in Harvester module, but required to exist across all of the modules, might cause some headaches with debugging or maintaining this in the future. Is there a way to only define this in one place and use across? Or something like, only have it in the Harvester module and use where necessary only if the var exists?

Reply:
Yeah so this was the beginnings of an effort to enable future updates which would ideally let us use different images for node(s) if we have a need for that, instead of relying on all of them to use the same image. Because the default is null this shouldn't interfere with any of the other modules. 

Because of the componentized approach of our tofu modules (specifically aroung the `generic/` modules), we have no way to directly tell the `test_environment` module what Harvester node image to use without this var.

Reply:
Yeah so this was the beginnings of an effort to enable future updates which would ideally let us use different images for node(s) if we have a need for that, instead of relying on all of them to use the same image. Because the default is null this shouldn't interfere with any of the other modules. 

Because of the componentized approach of our tofu modules (specifically aroung the `generic/` modules), we have no way to directly tell the `test_environment` module what Harvester node image to use without this var.